sources/packages/backend/apps/api/src/services/institution-location/institution-location.service.ts

```diff
@@ -170,16 +177,18 @@ export class InstitutionLocationService extends RecordDataModelService<Instituti
         "institutionLocation.id",
         "institutionLocation.institutionCode",
         "institutionLocation.primaryContact",
+        "institution.operatingName",
+        "institution.legalOperatingName",
+        "institution.primaryEmail",
       ])
+      .innerJoin("institutionLocation.institution", "institution")
       .where("institutionLocation.id = :locationId", {
         locationId: locationId,
       });
     if (institutionId) {
-      query
-        .innerJoin("institutionLocation.institution", "institution")
-        .andWhere("institution.id = :id", {
-          id: institutionId,
-        });
+      query.andWhere("institution.id = :id", {
```

Comment:
The `locationId` is mandatory and more specific than the `institutionId`, why do we need the `institutionId`?

Reply:
I see that this method is being called from multiple places and one of those is `getInstitutionLocation` called from `institution-location.institutions.controller.ts` which passes both location and institution id (seems to be for authorization purpose).